tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -175,11 +177,14 @@ def __init__(self, logger, browser, command_queue, no_timeout=False):
 
     def update_settings(self, test):
         browser_settings = self.browser.settings(test)
+
         restart_required = ((self.browser_settings is not None and
                              self.browser_settings != browser_settings) or
+                            self.pac != test.pac or
```

Comment:
Instead of storing `pac` on the class directly and checking this here, it should be part of the return value of `browser.settings(test)` for browsers that require a restart to change the proxy settings (for Firefox I'm pretty sure we can change the pref at runtime, so it should be part of the `test.environment` dict).